src/components/List/List.tsx

```diff
@@ -20,26 +21,63 @@ export default function List(props: React.PropsWithChildren<ListProps>) {
 
   const baseClass = "list";
 
-  let listTag;
+  let listTag,
+    errorText = "";
 
-  React.Children.map(children, function (child: React.ReactElement) {
-    if (child.type !== "li") {
-      throw new Error("Direct children of `List` should be `<li>`s");
-    }
-  });
-
-  if (type === ListTypes.Unordered) {
-    listTag = (
-      <ul id={id} className={bem(baseClass, modifiers, blockName, [className])}>
-        {children}
-      </ul>
-    );
-  } else if (type === ListTypes.Ordered) {
-    listTag = (
-      <ol id={id} className={bem(baseClass, modifiers, blockName, [className])}>
-        {children}
-      </ol>
-    );
+  switch (type) {
+    case "ol":
+      errorText = "Direct children of `List` (ordered) should be `<li>`s";
+      React.Children.map(children, function (child: React.ReactElement) {
+        if (child.type !== "li") {
+          throw new Error(errorText);
+        }
+      });
+      listTag = (
+        <ol
+          id={id}
+          className={bem(baseClass, modifiers, blockName, [className])}
+        >
+          {children}
+        </ol>
+      );
+      break;
+    case "ul":
+      errorText = "Direct children of `List` (unordered) should be `<li>`s";
+      React.Children.map(children, function (child: React.ReactElement) {
+        if (child.type !== "li") {
+          throw new Error(errorText);
+        }
+      });
+      listTag = (
+        <ul
+          id={id}
+          className={bem(baseClass, modifiers, blockName, [className])}
+        >
+          {children}
+        </ul>
+      );
+      break;
+    case "dl":
+      errorText =
+        "Direct children of `List` (definition) should be `<dt>`s or `<dd>`s";
+      React.Children.map(children, function (child: React.ReactElement) {
+        if (
+          child.type !== "dt" &&
+          child.type !== "dd" &&
+          child.type !== React.Fragment
+        ) {
+          throw new Error(errorText);
+        }
+      });
+      listTag = (
+        <dl id={id} className={bem("definition-list", modifiers, baseClass)}>
+          <Heading id="heading1" level={3}>
```

Comment:
I'm not sure this is semantically correct but check with Zach. I'm guessing the DL in figma has this heading in place already? Do all DLs need this? Because if not, it can be annoying to always render `Details` and only `Details` (if it's not a variable) -- but not a blocker for this PR.

Reply:
This will become an optional prop, I just haven't finished this part yet.